jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -101,9 +104,36 @@ public Throwable getThrowable() {
 	
 	private static native boolean isCheckpointAllowed();
 
-	private static native CRIUResult checkpointJVMImpl(String checkPointDir, boolean keepRunning, boolean shellJob, boolean extUnixSupport, int logLevel, String logFile);
+	private native CRIUResult checkpointJVMImpl();
 
-	private CRIUSupport() {}
+	/**
+	 * Constructs a new {@code CRIUSupport}.
+	 * 
+	 * The default CRIU dump options are:
+	 * {@code imagesDir} = null (must be set)
+	 * {@code leaveRunning} = false
+	 * {@code shellJob} = true
+	 * {@code extUnixSupport} = true
+	 * {@code logLevel} = 2
+	 * {@code logFile} = criu.log
+	 * {@code fileLocks} = false
+	 * {@code workDir} = imagesDir
+	 * 
+	 * @param imagesDir the directory that will hold the dump files as a java.nio.file.Path
+	 * @throws NullPointerException if imagesDir is null.
+	 * @throws SecurityException if no permission to access imagesDir.
```

Comment:
also if you dont have `CRIU_DUMP_PERMISSION`

Reply:
> Okay, this suggests that we need at least imageDir to be set. We should also throw in native code if imageDir is NULL
Can I assert? since the java code prevents it from being null

Reply:
yes, you can assert that.